packages/teams-js/src/private/externalAppAuthenticationForCEA.ts

```diff
@@ -0,0 +1,269 @@
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
+      getApiVersionTag(
+        externalAppAuthenticationTelemetryVersionNumber,
+        ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      ),
+      ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      [appId.toString(), conversationId, authTokenRequest.claims, authTokenRequest.silent],
+    );
+    if (error) {
+      throw error;
+    }
+  }
+
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform authentication using the given authentication parameters and then send the auth result to the application backend.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application backend to which the request and authentication response should be sent. This must be a UUID
+   * @param conversationId ConversationId To tell the bot what conversation the calls are coming from
+   * @param authenticateParameters Parameters for the authentication pop-up
+   * @throws InvokeError if the host encounters an error while authenticating
+   * @returns A promise that resolves from the application backend and rejects with InvokeError if the host encounters an error while authenticating
+   */
+  export async function authenticateWithOauth(
+    appId: AppId,
+    conversationId: string,
+    authenticateParameters: externalAppAuthentication.AuthenticatePopUpParameters,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    // Ask the parent window to open an authentication window with the parameters provided by the caller.
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
+      getApiVersionTag(
+        externalAppAuthenticationTelemetryVersionNumber,
+        ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithOauth,
+      ),
+      ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithOauth,
+      [
+        appId.toString(),
+        conversationId,
+        authenticateParameters.url.href,
+        authenticateParameters.width,
+        authenticateParameters.height,
+        authenticateParameters.isExternal,
+      ],
+    );
+    if (error) {
+      throw error;
+    }
+  }
+
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform authentication using the given authentication parameters and then resend the request to the application backend with the authentication result.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId ID of the application backend to which the request and authentication response should be sent. This must be a UUID
```

Comment:
>application backend [](http://example.com/codeflow?start=28&length=19)

Is this the appid of the bot? I'm not quite sure what "application backend" means here.

Reply:
This is the app ID of the third-party app on behalf of which Copilot is making a call

Reply:
FWIW that explanation would make for a clearer comment to me, anyway, e.g.,
```
@param appId App ID of the app upon whose behalf Copilot is requesting authentication. This must be a UUID.
```